lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
+        set: function(s) {
+          this._localTransform
+            .clear()
+            .scale(s)
+            .rotate(this._rotation)
+            .translate(this._offset);
+          this._onTransformChanged();
+        }.bind(this)
+      }
+    });
+
+    this._onTransformChanged();
+  };
+
+  /**
+   * Update this collider based on the properties of a parent Sprite.
+   * Descendant classes should override this method to adopt the dimensions
+   * of the sprite if `getsDimensionsFromSprite` is true.
+   * @method updateFromSprite
+   * @param {Sprite} sprite
+   * @see p5.CollisionShape.prototype.getsDimensionsFromSprite
+   */
+  p5.CollisionShape.prototype.updateFromSprite = function(sprite) {
+    this.setParentTransform(sprite);
+  };
+
+  /**
+   * Update this collider's parent transform, which will in turn adjust its
+   * position, rotation and scale in world-space and recompute cached values
+   * if necessary.
+   * If a Sprite is passed as the 'parent' then a new transform will be computed
+   * from the sprite's position/rotation/scale and used.
+   * @method setParentTransform
+   * @param {p5.Transform2D|Sprite} parent
+   */
+  p5.CollisionShape.prototype.setParentTransform = function(parent) {
+    if (parent instanceof Sprite) {
+      this._parentTransform
+        .clear()
+        .scale(parent.scale) // TODO: ScaleX and ScaleY?
+        .rotate(radians(parent.rotation))
+        .translate(parent.position);
+    } else if (parent instanceof p5.Transform2D) {
+      this._parentTransform = parent.copy();
+    } else {
+      throw new TypeError('Bad argument to setParentTransform: ' + parent);
+    }
+    this._onTransformChanged();
+  };
+
+  /**
+   * Recalculate cached properties, relevant vectors, etc. when at least one
+   * of the shape's transforms changes.  The base CollisionShape (and CollisionPoint)
+   * only need to recompute the shape's center, but other shapes may need to
+   * override this method and do additional recomputation.
+   * @method _onTransformChanged
+   * @protected
+   */
+  p5.CollisionShape.prototype._onTransformChanged = function() {
+    // Recompute internal properties from transforms
+
+    // Rotation in local space
+    this._rotation = this._localTransform.getRotation();
+
+    // Scale in local space
+    this._scale = this._localTransform.getScale();
+
+    // Offset in local-space
+    this._offset
+      .set(0, 0)
+      .transform(this._localTransform);
+
+    // Center in world-space
+    this._center
+      .set(this._offset.x, this._offset.y)
+      .transform(this._parentTransform);
+  };
+
+  /**
+   * Compute the smallest movement needed to move this collision shape out of
+   * another collision shape.  If the shapes are not overlapping, returns a
+   * zero vector to indicate that no displacement is necessary.
+   * @method collide
+   * @param {p5.CollisionShape} other
+   * @return {p5.Vector}
+   */
+  p5.CollisionShape.prototype.collide = function(other) {
```

Comment:
Finds minimum displacement using the Separating Axis Theorem.  If you've never read [N Tutorial A from metanet](http://www.metanetsoftware.com/technique/tutorialA.html), stop and do that now - it will make it a lot easier to understand this method.